tests/integration/metrics/column/test_nonnull_count.py

```diff
@@ -0,0 +1,45 @@
+import pandas as pd
+
+from great_expectations.datasource.fluent.interfaces import Batch
+from great_expectations.metrics.column.aggregate_nonnull_count import (
+    ColumnAggregateNonNullCount,
+    ColumnAggregateNonNullCountResult,
+)
+from tests.integration.conftest import parameterize_batch_for_data_sources
+from tests.metrics.conftest import (
+    PANDAS_DATA_SOURCES,
+    SPARK_DATA_SOURCES,
+    SQL_DATA_SOURCES,
+)
+
+STRING_COLUMN_NAME = "whatevs"
+DATA_FRAME = pd.DataFrame(
+    {
+        STRING_COLUMN_NAME: ["a", None, "c", "d", None],
+    },
+    dtype="object",
+)
+
+
+class TestColumnAggregateNonNullCount:
+    @parameterize_batch_for_data_sources(
+        data_source_configs=SQL_DATA_SOURCES + PANDAS_DATA_SOURCES,
+        data=DATA_FRAME,
+    )
+    def test_success(self, batch_for_datasource: Batch) -> None:
+        metric = ColumnAggregateNonNullCount(column=STRING_COLUMN_NAME)
+        metric_result = batch_for_datasource.compute_metrics(metric)
+
+        assert isinstance(metric_result, ColumnAggregateNonNullCountResult)
+        assert metric_result.value == 3
+
+    @parameterize_batch_for_data_sources(
+        data_source_configs=SPARK_DATA_SOURCES,
+        data=DATA_FRAME,
+    )
+    def test_spark(self, batch_for_datasource: Batch) -> None:
+        metric = ColumnAggregateNonNullCount(column=STRING_COLUMN_NAME)
+        metric_result = batch_for_datasource.compute_metrics(metric)
+
+        assert isinstance(metric_result, ColumnAggregateNonNullCountResult)
+        assert metric_result.value == 3
```

Comment:
I think you can get rid of this and add the `+ SPARK_DATA_SOURCES` to the `data_source_configs` in `test_success`. In some cases we have to break out certain datasources because they might ignore nulls in row counts or do something else different than the other data sources but that doesn't look to be the case here.

Also, could you add 3 more tests:
* One where every entry is null
* One where no entry is null
* An empty dataset